app/uk/gov/hmrc/emcstfe/controllers/GetSubmissionFailureMessageController.scala

```diff
@@ -17,25 +17,42 @@
 package uk.gov.hmrc.emcstfe.controllers
 
 import play.api.libs.json.Json
-import play.api.mvc.{Action, AnyContent, ControllerComponents}
+import play.api.mvc.{Action, AnyContent, ControllerComponents, Result}
 import uk.gov.hmrc.emcstfe.controllers.actions.{AuthAction, AuthActionHelper}
 import uk.gov.hmrc.emcstfe.models.request.GetSubmissionFailureMessageRequest
+import uk.gov.hmrc.emcstfe.models.response.getSubmissionFailureMessage.GetSubmissionFailureMessageResponse
+import uk.gov.hmrc.emcstfe.repositories.CreateMovementUserAnswersRepository
 import uk.gov.hmrc.emcstfe.services.GetSubmissionFailureMessageService
 import uk.gov.hmrc.play.bootstrap.backend.controller.BackendController
 
 import javax.inject.{Inject, Singleton}
-import scala.concurrent.ExecutionContext
+import scala.concurrent.{ExecutionContext, Future}
 
 @Singleton
 class GetSubmissionFailureMessageController @Inject()(cc: ControllerComponents,
                                                       service: GetSubmissionFailureMessageService,
+                                                      createMovementUserAnswersRepository: CreateMovementUserAnswersRepository,
                                                       override val auth: AuthAction
                                                      )(implicit ec: ExecutionContext) extends BackendController(cc) with AuthActionHelper {
 
   def getSubmissionFailureMessage(exciseRegistrationNumber: String, messageId: String): Action[AnyContent] = authorisedUserRequest(exciseRegistrationNumber) { implicit request =>
-    service.getSubmissionFailureMessage(GetSubmissionFailureMessageRequest(exciseRegistrationNumber = exciseRegistrationNumber, messageId = messageId)).map {
-      case Left(value) => InternalServerError(Json.toJson(value))
-      case Right(value) => Ok(Json.toJson(value))
+    service.getSubmissionFailureMessage(GetSubmissionFailureMessageRequest(exciseRegistrationNumber = exciseRegistrationNumber, messageId = messageId)).flatMap {
+      case Left(error) => Future(InternalServerError(Json.toJson(error)))
+      case Right(response) =>
+        val correlationID = response.ie704.header.correlationIdentifier.getOrElse("")
+        if(correlationID.startsWith("PORTAL")) {
+          Future(returnResponse(response, isTFESubmission = true))
+        } else {
+          createMovementUserAnswersRepository.get(exciseRegistrationNumber, correlationID).map {
+            optDraftMovement => {
+              returnResponse(response, isTFESubmission = optDraftMovement.isDefined)
+            }
+          }
+        }
```

Comment:
Feel like we should refactor out the `.getOrElse` here. Otherwise it's going to call the `createMOvementUserAnswersRepository` with `""` when it doesn't need to as it won't return anything.

Would it be better to pattern match on the value to then decide what to do? Like:

```suggestion
      response.ie704.header.correlationIdentifier match {
          case Some(correlationId) if correlationId.startsWith("PORTAL") =>
            Future(returnResponse(response, isTFESubmission = true))
          case None =>
            Future(returnResponse(response, isTFESubmission = false))
          case Some(correlationId) =>
            createMovementUserAnswersRepository.get(exciseRegistrationNumber, correlationId).map { optDraftMovement =>
              returnResponse(response, isTFESubmission = optDraftMovement.isDefined)
            }
        }
```